scripts/zones/Spire_of_Dem/mobs/Progenerator.lua

```diff
@@ -1,29 +1,33 @@
 -----------------------------------
 -- Area: Spire of Dem
 --  Mob: Progenerator
--- TODO: Verify cmbDelay
 -----------------------------------
 mixins = { require('scripts/mixins/families/empty_terroanima') }
 -----------------------------------
 ---@type TMobEntity
 local entity = {}
 
+entity.onMobInitialize = function(mob)
+    mob:addImmunity(xi.immunity.DARK_SLEEP)
+    mob:addImmunity(xi.immunity.LIGHT_SLEEP)
+end
+
 entity.onMobSpawn = function(mob)
-    mob:setLocalVar('maxBabies', 4)
-    mob:addMod(xi.mod.TRIPLE_ATTACK, 10)
-    mob:addMod(xi.mod.DEFP, 35)
+    mob:setMod(xi.mod.DEFP, 35)
+    mob:setMod(xi.mod.TRIPLE_ATTACK, 10)
+    mob:setMod(xi.mod.STORETP, 62)
+    mob:setMobMod(xi.mobMod.WEAPON_BONUS, 15)
 end
 
 entity.onMobWeaponSkillPrepare = function(mob, target)
-    local fission = 755
-    local random = math.random(1, 100)
-
-    if mob:getHPP() <= 50 then
-        if random <= 60 then
-            return fission
-        else
-            return 0
-        end
+    if mob:getHPP() <= 35 then
+        local tpMoves =
+        {
+            xi.mobSkill.FISSION,
+            xi.mobSkill.SPIRIT_ABSORPTION_2,
+            xi.mobSkill.VANITY_DRIVE_2,
+        }
+        return tpMoves[math.random(#tpMoves)]
```

Comment:
What happens when hpp is above 35%?

Reply:
They use their normal TP move skill list.